src/Engine.php

```diff
@@ -161,6 +161,23 @@ public function search(string $query, string $indexName, array $searchParams): a
         return $this->client->index($indexName)->rawSearch($query, $searchParams);
     }
 
+    /**
+     * Search the index and returns the document identifier.
+     *
+     * @param string                          $query
+     * @param string                          $indexName
+     * @param array<string, int|string|array> $searchParams
+     *
+     * @return array<string, int|string|array>
```

Comment:
Can this method return an array<string, array> ? This would mean the ID itself would be an array.